### PR TITLE
Changed bconsole, filedaemon, storage & traymonitor State files

### PR DESCRIPTION
**Pull request by pkleanthous**
* Pin package versions
  * Separate the installation of core packages and plugins

### CODE REVIEW

File: bareos/bconsole.sls [hunk 1]
```diff
@@ -31,5 +32,5 @@ bareos_bconsole:
     - user: root
     - group: root
     - require:
-      - pkg: bareos_bconsole
+      - pkg: bareos-bconsole
```
Comment (javierbertoli) [inline bareos/bconsole.sls]: You changed here the dependency on the *resource* to a dependency on the *real package*. This way, if the package name changes for whatever reason, the dependency will fail. Perhaps it's better if you keep the depedency on the resource, like

```yaml
      - pkg: install_bconsole_package
```

so we depend on the resource success instead that depending on the existance of certain package? This abstracts the formula for the name the distros give to packages (ie, bconsole, bareos-console, bareos-bconsole, etc.)

File: bareos/filedaemon.sls [hunk 2]
```diff
@@ -31,14 +40,14 @@ bareos_filedaemon:
     - user: root
     - group: root
     - require:
-      - pkg: bareos_filedaemon
+      - pkg: bareos-filedaemon
```
Comment (javierbertoli) [inline bareos/filedaemon.sls]: Same as above, can you change it to depend on `install_fd_package`?

File: bareos/filedaemon.sls [hunk 3]
```diff
@@ -31,14 +40,14 @@ bareos_filedaemon:
...
 {% endif %}
 
+bareos_fd_service:
   service.running:
     - name: {{ bareos.filedaemon.service }}
     - enable: true
     - require:
-      - pkg: bareos_filedaemon
-
+      - pkg: bareos-filedaemon
```
Comment (javierbertoli) [inline bareos/filedaemon.sls]: Same as above, can you change it to depend on `install_fd_package`?

File: bareos/storage.sls [hunk 4]
```diff
@@ -31,14 +40,14 @@ bareos_storage:
     - user: root
     - group: root
     - require:
-      - pkg: bareos_storage
+      - pkg: bareos-storage
```
Comment (javierbertoli) [inline bareos/storage.sls]: Same as above, can you change it to depend on `install_storage_package`?

File: bareos/storage.sls [hunk 5]
```diff
@@ -31,14 +40,14 @@ bareos_storage:
...
 {% endif %}
 
+bareos_storage_service:
   service.running:
     - name: {{ bareos.storage.service }}
     - enable: true
     - require:
-      - pkg: bareos_storage
-
+      - pkg: bareos-storage
```
Comment (javierbertoli) [inline bareos/storage.sls]: Same as above, can you change it to depend on `install_storage_package`?

File: bareos/traymonitor.sls [hunk 6]
```diff
@@ -3,22 +3,23 @@
...
 {% set require_password = ['director'] %}
-{% set pkgs = [bareos.traymonitor.pkg] %}
 
 {% if bareos.use_upstream_repo %}
 include:
   - bareos.repo
 {% endif %}
 
-bareos_traymonitor:
+install_trymon_package:
```
Comment (javierbertoli) [inline bareos/traymonitor.sls]: mispelled 'traymon', missing an 'a' 😄

File: bareos/traymonitor.sls [hunk 7]
```diff
@@ -3,22 +3,23 @@
...
-    - pkgs: {{ pkgs }}
+    - name: {{ bareos.traymonitor.pkg }}
+    - version: {{ bareos.version }}
     {% if bareos.use_upstream_repo %}
     - require:
       - pkgrepo: bareos_repo
     {% endif %}
 
 {% if tm_config != {} %}
+bareos_trymon_cfg_file:
```
Comment (javierbertoli) [inline bareos/traymonitor.sls]: Here too, missing an 'a'

File: bareos/traymonitor.sls [hunk 8]
```diff
@@ -31,5 +32,5 @@ bareos_traymonitor:
     - user: root
     - group: root
     - require:
-      - pkg: bareos_traymonitor
+      - pkg: bareos-traymonitor
```
Comment (javierbertoli) [inline bareos/traymonitor.sls]: Same as with the other deps, can you change it to depend on `install_traymon_package`?

**Comment by pkleanthous** [comment]
@javierbertoli Thanks for your comments all of them are valid so I pushed the relevant changes.

**Comment by javierbertoli** [comment]
Awesome work, @pkleanthous! Thanks!

**Pull request merged**